libnavigation-core/src/main/java/com/mapbox/navigation/core/OnboardRouterFiles.kt

```diff
@@ -5,22 +5,20 @@ import com.mapbox.base.common.logger.Logger
 import com.mapbox.base.common.logger.model.Message
 import com.mapbox.base.common.logger.model.Tag
 import com.mapbox.navigation.base.options.OnboardRouterOptions
-import com.mapbox.navigation.utils.internal.ThreadController
 import java.io.File
-import kotlinx.coroutines.withContext
 
 internal class OnboardRouterFiles(
     val applicationContext: Context,
     val logger: Logger
 ) {
 
-    suspend fun absolutePath(options: OnboardRouterOptions): String? = withContext(ThreadController.IODispatcher) {
+    fun absolutePath(options: OnboardRouterOptions): String? {
```

Comment:
it's good to know that this is causing the sync issue. but i don't think the solution is to access the file system on the main thread - that's trading issues

Reply:
If we merge this, can you cut a ticket to make a proper fix for this?

Accessing the file system can be extremely slow on automotive devices for example. It may not be seen when you're on your mobile phone. What if the operating system is on an external hard-drive? This call can take a second

Reply:
> can you cut a ticket to make a proper fix for this?

Follow up ticket 👉 https://github.com/mapbox/mapbox-navigation-android/issues/3417